### PR TITLE
reduce image creation and container startup times

### PR DESCRIPTION
**Pull request by trevj**
The goal here is to dramatically reduce image creation and container startup times.

The two things I've done are:
 * push zork download/build to image creation time
 * be selective about what to install in the image

On my workstation (admittedly a fast and well-connected system) I'm able to spin up a cloud system from scratch in **<4 mins**.

Also a little tidy-up in here, e.g. installing by NPM is now the default, eliminating a bunch of `-n` options.

Two followups:
 * distribute uproxy-lib as a gzip, associated with the release (npm requires **>150MB** of dependenices, WTF)
 * reduce the number of dependencies in the SSHD image (we're installing npm just so we can generate a v2-style invite...not necessary for most users).

### CODE REVIEW

**Comment by trevj** [comment]
@mollyling FYI

File: testing/integration/test/bin/load-zork.sh [hunk 1]
```diff
@@ -3,77 +3,37 @@
...
-  else
-    npm install -g bower grunt-cli
-    echo git clone $CLONEARGS $CLONESRC
-    git clone $CLONEARGS $CLONESRC
-    cd uproxy-lib
-    ./setup.sh install
-    grunt zork
-  fi
+if [ "$RUNVNC" = true ]; then
+    x11vnc -display :10 -forever &
```
Comment (soycode) [inline testing/integration/test/bin/load-zork.sh]: Nit, 2 space indent.

File: testing/run-scripts/image_make.sh [hunk 2]
```diff
@@ -1,21 +1,85 @@
...
+  case $opt in
+    g) GIT=true ;;
+    b) BRANCH="-b $OPTARG" ;;
+    p) PREBUILT=true ;;
+    *) usage ;;
+  esac
+done
+shift $((OPTIND-1))
+
+if [ "$GIT" = true ] && [ "$PREBUILT" = true ]
```
Comment (soycode) [inline testing/run-scripts/image_make.sh]: May also be worth checking if -b is specified but -g isn't (doesn't cause an error but the -b is useless in that case).

File: testing/integration/test/bin/load-zork.sh (hunk 1 above)
Comment (trevj) [inline testing/integration/test/bin/load-zork.sh]: Fixed.

File: testing/run-scripts/image_make.sh (hunk 2 above)
Comment (trevj) [inline testing/run-scripts/image_make.sh]: Good idea! Added.

**Comment by trevj** [comment]
Thanks! PTAL.

**Comment by mollyling** [comment]
Hi, quick question: who's building the docker image?  I thought `docker build` was part of the setup script.

**Comment by soycode** [comment]
Tested successfully on Linode, took ~7mins. Changes look good, re: trying to further cut down on npm overhead, that'd definitely be nice but it may be tricky to ensure it all still works (the 150mb it pulls in is probably all the dependency-of-dependency stuff - though if we cut build specific dependencies that may help).

**Comment by trevj** [comment]
Yes, it's dependencies. We only install npm to use jsurl which seems like massive overkill - we can probably install that on-demand, for manual installs.

Taking your "changes look good" as a :+1:, thanks!

**Comment by trevj** [comment]
@mollyling `image_make.sh` still invokes `docker build`. Invoked by both `run_cloud.sh` and `run_pair.sh`.

**Pull request merged**